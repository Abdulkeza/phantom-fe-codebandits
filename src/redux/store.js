import { configureStore } from '@reduxjs/toolkit';
import { counterReducer } from './reducers/counterReducer.js';
import { updateProfileReducer } from './reducers/updateProfileReducer.js';
import { profileReducer } from './reducers/profileReducer.js';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    profileUpdates: updateProfileReducer,
    profileInfo: profileReducer
  }
});
