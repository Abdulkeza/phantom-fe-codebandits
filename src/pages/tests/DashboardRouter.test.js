import React from 'react';
import { Provider} from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { store } from '../../redux/store.js';
import DashboardRouter from '../DashboardRouter.js';

describe('DashboardRouter tests', () => {
  it('it should render DashboardRouter', () => {
    const elem = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <DashboardRouter />
          </MemoryRouter>
        </Provider>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
});
