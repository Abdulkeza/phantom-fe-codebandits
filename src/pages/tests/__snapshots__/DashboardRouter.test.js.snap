// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardRouter tests it should render DashboardRouter 1`] = `
<div
  className="flex z-0 h-full font-sans antialiased bg-grey-lightest w-full"
>
  <div
    className="relative z-10"
  >
    <main
      className="hidden md:flex lg:flex xl:flex flex text-sm"
    >
      <aside
        className="border-r-2 h-full sticky top-0 px-8 py-4 bg-gray-100 transition-transform"
      >
        <div
          className=" mb-6"
        >
          <span
            className="rounded-full mx-auto flex items-center justify-center h-12 w-12 mb-5"
          >
            <img
              alt="Image"
              className="rounded-full object-cover h-12 w-12"
              src="profileImg.jpg"
            />
          </span>
          <div
            className="text-center font-black text-md"
          >
            <h1>
              NGABO Patrick
            </h1>
            <div>
              <h1
                className="font-bold"
              >
                Operator
              </h1>
            </div>
          </div>
        </div>
        <div>
          <h2
            className="mb-2 text-lg font-black"
          >
            Account
          </h2>
          <ul
            className="mb-2"
          >
            <div
              className="flex justify-center items-center hover:bg-gray-200 py-1"
            >
              <img
                alt="icon"
                className="w-6"
                src="dashboardIcon.png"
              />
              <a
                className="py-1 px-2 font-bold my-1  cursor-pointer w-full"
                href="/account"
                onClick={[Function]}
              >
                <li>
                  Dashboard
                </li>
              </a>
            </div>
            <div
              className="flex justify-center items-center hover:bg-gray-200 py-1"
            >
              <img
                alt="icon"
                className="w-6"
                src="manageIcon.png"
              />
              <a
                className="py-1 px-2 font-bold my-1  cursor-pointer w-full"
                href="/management"
                onClick={[Function]}
              >
                <li>
                  Management
                </li>
              </a>
            </div>
            <div
              className="flex justify-center items-center hover:bg-gray-200 py-1"
            >
              <img
                alt="icon"
                className="w-6"
                src="profileIcon.png"
              />
              <a
                className="py-1 px-2 font-bold my-1  cursor-pointer w-full"
                href="/profile"
                onClick={[Function]}
              >
                <li>
                  Profile
                </li>
              </a>
            </div>
            <div
              className="flex justify-center items-center hover:bg-gray-200 py-1"
            >
              <img
                alt="icon"
                className="w-6"
                src="notificationIcon.png"
              />
              <a
                aria-current="page"
                className="py-1 px-2 font-bold my-1  cursor-pointer w-full active"
                href="/#p"
                onClick={[Function]}
              >
                <li>
                  Notifications
                </li>
              </a>
            </div>
            <div
              className="flex justify-center items-center hover:bg-gray-200 py-1"
            >
              <img
                alt="icon"
                className="w-6"
                src="supportIcon.png"
              />
              <a
                aria-current="page"
                className="py-1 px-2 font-bold my-1  cursor-pointer w-full active"
                href="/#p"
                onClick={[Function]}
              >
                <li>
                  Support
                </li>
              </a>
            </div>
            <div
              className="flex justify-center items-center hover:bg-gray-200 py-1"
            >
              <img
                alt="icon"
                className="w-6"
                src="settingIcon.png"
              />
              <a
                aria-current="page"
                className="py-1 px-2 font-bold my-1  cursor-pointer w-full active"
                href="/#p"
                onClick={[Function]}
              >
                <li>
                  Settings
                </li>
              </a>
            </div>
          </ul>
          <div>
            <button
              className="bg-red px-6 py-1 hover:bg-red-600 bg-blue-500 hover:bg-blue-600 text-white font-bold py-2 px-4 rounded-xl m-2 "
              onClick={[Function]}
              type="submit"
            >
              Logout
            </button>
          </div>
        </div>
      </aside>
    </main>
    <div
      className="flex"
    >
      <button
        className="md:hidden lg:hidden xl:hidden"
        onClick={[Function]}
        type="button"
      >
        <div>
          <div
            className="ml-2 bg-black h-1 w-7 bg-primary mt-1"
          />
          <div
            className="ml-2 bg-black h-1 w-7 bg-primary mt-1"
          />
          <div
            className="ml-2 bg-black h-1 w-7 bg-primary mt-1"
          />
        </div>
      </button>
    </div>
  </div>
  <div
    className=" w-4/5 float-right h-full p-4 top-8 right-2 "
  >
    <main
      className="font-bold flex h-full w-full items-center justify-center"
    >
      <h1
        className="text-3xl pb-4 font-bold mb-4"
      >
        Nothing to see here.
      </h1>
      <a
        className="text-white rounded-md bg-primary py-1 px-4 m-3"
        href="/dashboard/main"
        onClick={[Function]}
      >
        Main Page
      </a>
    </main>
  </div>
</div>
`;
